Deployment

elbotho opened the pull request:
- https://github.com/serlo/api.serlo.org/pull/1399 
- https://github.com/serlo/api.serlo.org/pull/1403
- https://github.com/serlo/api.serlo.org/pull/1404
- https://github.com/serlo/api.serlo.org/pull/1405
- https://github.com/serlo/api.serlo.org/pull/1407
- https://github.com/serlo/api.serlo.org/pull/1409
- https://github.com/serlo/api.serlo.org/pull/1392
- https://github.com/serlo/api.serlo.org/pull/1411
- https://github.com/serlo/api.serlo.org/pull/1412
- https://github.com/serlo/api.serlo.org/pull/1415
- https://github.com/serlo/api.serlo.org/pull/1413
- https://github.com/serlo/api.serlo.org/pull/1416

AndreasHuber committed:
refactor(AllThreadsQuery): use new database.ts

AndreasHuber committed:
fix(threads): remove unused import

AndreasHuber committed:
refactor(setDescription): move SQL query into resolver

AndreasHuber committed:
fix(setDescription): fix some errors/warnings

AndreasHuber committed:
style(setDescription): run prettier

AndreasHuber committed:
refactor(setDescription): remove from Serlo model

AndreasHuber committed:
refactor(database): remove the model/database.ts

AndreasHuber committed:
refactor(experiments): use new database.ts instead of model/database.ts

AndreasHuber committed:
style(experiments): run prettier

AndreasHuber committed:
test(all-threads): use new database.ts instead of model/database.ts

AndreasHuber committed:
test(all-threads): run prettier

AndreasHuber committed:
test(all-threads): change import order

elbotho committed:
Merge pull request #1399 from serlo/production

Merge pull request #1382 from serlo/staging

AndreasHuber committed:
Merge branch 'staging' into use-new-database.ts

kulla committed:
refactor: Remove HasNextPageInfo + Edges

Fixes #1394
Fixes https://github.com/serlo/api.serlo.org/issues/1402

kulla committed:
refactor: Simplify resolveConnection()

kulla committed:
test: Use toThrow()

kulla committed:
Merge pull request #1403 from serlo/1394-merge-graphql-types-hasnextpageinfo-and-pageinfo

refactor: Remove HasNextPageInfo + Edges

elbotho commented on packages/server/src/schema/notification/types.graphql:
```diff
@@ -17,19 +17,15 @@ interface AbstractNotificationEvent {
     after: String
-    before: String
     first: Int
-    last: Int
```
@kulla removing `last` breaks the frontend and I don't have time to fix it right now (also means that staging and lenabi are broken). It's currently used in the events query.

kulla replied:
Okay, I am working on it

kulla committed:
refactor: Move InvalidCurrentValueError to ~/errors

kulla committed:
Format ~/errors

kulla committed:
refactor: Change InvalidCurrentValueError tp GraphQLError

AndreasHuber committed:
Update packages/server/src/schema/uuid/user/resolvers.ts

Co-authored-by: Stephan Kulla <git.mail@kulla.me>

kulla committed:
refactor: Use Sentry directely to avoid cyclic imports

kulla committed:
Merge pull request #1404 from serlo/SNGqp-kulla-2024-04-18-12-28

refactor: Small changes to avoid cyclic imports

kulla committed:
Revert "refactor: Remove HasNextPageInfo + Edges"

kulla committed:
Merge pull request #1405 from serlo/revert-1403-1394-merge-graphql-types-hasnextpageinfo-and-pageinfo

Revert "refactor: Remove HasNextPageInfo + Edges"

kulla committed:
Revert "Revert "refactor: Remove HasNextPageInfo + Edges""

kulla committed:
fix: Reintroduce "last" arg in events()

AndreasHuber committed:
fix(setDescription): try to fix errors with proposed alternative to remove cache entry

kulla committed:
Merge pull request #1406 from serlo/revert-1405-revert-1403-1394-merge-graphql-types-hasnextpageinfo-and-pageinfo

Revert "Revert "refactor: Remove HasNextPageInfo + Edges""

AndreasHuber committed:
fix(setDescription): put the original setCache in the resolver for now

AndreasHuber committed:
style(setDescription): format

kulla committed:
refactor: Move error-event.ts

kulla committed:
refactor(error-event): Update deprecated type variables

kulla committed:
refactor: Move auth-services to ~/context

kulla committed:
refactor: Remove Environment interface

kulla committed:
test: Add `authServices` to context

kulla committed:
Merge pull request #1407 from serlo/2wHV6-kulla-2024-04-18-14-00

refactor: Remove Environment and move all necesary services into context

AndreasHuber committed:
refactor(setDescription): remove now obsolete decoder from database-layer.ts

AndreasHuber committed:
Merge branch 'staging' into use-new-database.ts

kulla committed:
fix(swr-queue): Add createKratosModel to modelFactories

kulla committed:
Merge pull request #1409 from serlo/1408-kratos-model-is-not-updated-per-swr-queue-worker

fix(swr-queue): Add createKratosModel to modelFactories

kulla committed:
refactor: Delete __tests__/internals/data-sources.ts

AndreasHuber committed:
Merge pull request #1392 from serlo/use-new-database.ts

refactor(AllThreadsQuery): use new database.ts

AndreasHuber committed:
Merge pull request #1411 from serlo/pUTtu-kulla-2024-04-18-15-25

refactor: Delete __tests__/internals/data-sources.ts

kulla committed:
refactor: Merge `lock-manager.ts` into `cache/index.ts`

kulla committed:
refactor: Refactor cache/index.ts

kulla committed:
refactor: Move cache to root level

kulla committed:
Merge pull request #1412 from serlo/Dztxh-kulla-2024-04-18-15-32

refactor: Move cache code to root level + refactor it

kulla committed:
refactor: Remove log() utility function

kulla committed:
fix: Fix getPayload() of UuidResolver

kulla committed:
test: Test getKey() and getPayload() of resolvers

kulla committed:
refactor: "resolverNameForErrorMessage" => "name"

kulla committed:
Merge pull request #1415 from serlo/kJ8Py-kulla-2024-04-18-16-06

test: Test getKey() and getPayload() of new resolvers

hugotiburtino committed:
Merge pull request #1413 from serlo/Ykuz5-kulla-2024-04-18-15-50

refactor: Remove log() utility function

hugotiburtino committed:
feat(kratos): do not depend on userId to sync all accounts

hugotiburtino committed:
test(kratos): adapt test to new behavior in register endpoint

hugotiburtino committed:
style(test): disable eslint error regarding unbound method

hugotiburtino committed:
Merge pull request #1416 from serlo/kratos-registration-fix

fix: Change kratos register endpoint to always sync accounts

dependabot[bot] committed:
build(deps-dev): bump @apollo/server from 4.10.2 to 4.10.4

Bumps [@apollo/server](https://github.com/apollographql/apollo-server/tree/HEAD/packages/server) from 4.10.2 to 4.10.4.
- [Release notes](https://github.com/apollographql/apollo-server/releases)
- [Changelog](https://github.com/apollographql/apollo-server/blob/main/packages/server/CHANGELOG.md)
- [Commits](https://github.com/apollographql/apollo-server/commits/@apollo/server@4.10.4/packages/server)

---
updated-dependencies:
- dependency-name: "@apollo/server"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

hugotiburtino committed:
Merge pull request #1426 from serlo/dependabot/npm_and_yarn/apollo/server-4.10.4

build(deps-dev): bump @apollo/server from 4.10.2 to 4.10.4
